src/ui/App.tsx

```diff
@@ -52,6 +57,15 @@ const App = () => {
     setShowToast(toastMsg !== undefined);
   }, [currentOperation, toastMsg]);
 
+  useEffect(() => {
+    if (
+      authentication.userName?.length === 0 &&
```

Comment:
This works (so leave it) - but just to note that if `userName` is `undefined | null`, `userName?.length === 0` will always be `false` which might give you the illusion that it's set.

But in this case I can see you always set it to `""` if its not in preferences, so it's OK, just a note